web/ui/dashboard/src/app/services/http/http.service.ts

```diff
@@ -99,4 +95,40 @@ export class HttpService {
 			}
 		});
 	}
+
+	async getRefreshToken() {
+		try {
+			const refreshedTokens = await this.request({
+				url: `/auth/token/refresh`,
+				method: 'post',
+				body: this.authDetails()
+			});
+			localStorage.setItem('CONVOY_AUTH_TOKENS', JSON.stringify(refreshedTokens.data));
+			window.location.reload();
+		} catch {
+			this.initiateLogout();
+		}
+	}
+
+	checkIfTokenIsExpired() {
+		const currentTime = new Date();
+		const tokenExpiryTime = this.jwtHelper.getTokenExpirationDate(this.authDetails().access_token);
+		if (tokenExpiryTime) {
+			const expiryPeriodInSeconds = differenceInSeconds(tokenExpiryTime, currentTime);
+			if (expiryPeriodInSeconds <= 180) this.getRefreshToken();
```

Comment:
This might lead to a recursion @Oluwadaminiola. If the expiry period is less than 180s, you're making a request with the `request` method again to get the refresh token, which will again run the `checkIfTokenIsExpired` method.

Will this continue in circles, or am I reading it wrong?

Reply:
actually, because a refresh occurs whenever the `getRefreshToken()` method is called, the `expiryPeriodInSeconds` resets

Reply:
so a recursion doesn't happen